wezterm-ssh/src/config.rs

```diff
@@ -175,7 +175,13 @@ impl ParsedConfigFile {
         groups: &mut Vec<MatchGroup>,
         loaded_files: &mut Vec<PathBuf>,
     ) {
-        match filenamegen::Glob::new(&pattern) {
+        let home = dirs_next::home_dir().unwrap_or_default();
```

Comment:
Thanks for diving in!

* Please make the `home_dir` resolution only trigger when we observe the `~/` prefix.
* I'm not clear on why the cwd overrides the caller-supplied value for either the `~` or `/` case; that's something that would benefit from some explanation in the code with comments.
* Please add unit tests for these new cases! You can find various examples towards the bottom of this file